### PR TITLE
[Apps] Redesign webhooks

### PR DESCRIPTION
**Pull request by lolimay**
depends on: https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/168

**Preview:**
image

**Test App:** https://github.com/lolimay/frvr-games-rocketchat-wrapper/blob/master/basketball-beta/BasketballBetaApp.ts

### CODE REVIEW

**Comment by lolimay** [comment]
Because of the [Rate Limiter](https://rocket.chat/docs/developer-guides/rest-api/rate-limiter/#rate-limiter) policy, if the user opens and closes an external component too quickly, the API requests will be blocked by Rocket.Chat and the events could be missing which will not be sent to apps successfully 😔

image

@d-gubert @graywolf336 Do you have any ideas to fix this problem? Thanks!

**Comment by graywolf336** [comment]
@lolimay create a new endpoint for you that you can control within the scope of the apps api.

**Comment by lolimay** [comment]
> @lolimay create a new endpoint for you that you can control within the scope of the apps api.

Thanks! I'll have a try :)ax

File: app/apps/server/communication/rest.js [hunk 1]
```diff
@@ -268,6 +270,54 @@ export class AppsRestApi {
...
+					&& ['IPostExternalComponentOpened', 'IPostExternalComponentClosed'].includes(this.bodyParams.event)
+				) {
+					let result;
+					try {
+						const { event } = this.bodyParams;
+						const { externalComponent } = this.bodyParams;
+
+						result = Apps.getBridges().getListenerBridge().externalComponentEvent(event, externalComponent);
+					} catch (e) {
+						orchestrator.getRocketChatLogger().error(`Error triggering exxternal components' events ${ e.response.data }`);
```
Comment (d-gubert) [inline app/apps/server/communication/rest.js]:
```suggestion
						orchestrator.getRocketChatLogger().error(`Error triggering external components' events ${ e.response.data }`);
```

**Pull request merged**